pandas/core/arrays/datetimelike.py

```diff
@@ -371,16 +388,15 @@ def _add_delta_td(self, other):
         inc = delta_to_nanoseconds(other)
         new_values = checked_add_with_arr(self.asi8, inc,
                                           arr_mask=self._isnan).view('i8')
-        if self.hasnans:
-            new_values[self._isnan] = iNaT
+        new_values = self._maybe_mask_results(new_values, fill_value=iNaT)
```

Comment:
why doesn't ``_maybe_mask_results`` have an explict ``iNaT`` as the fill value? that is almost always the appropriate one, relying on subclass writer to be correct about this seems wrong.

Reply:
No idea.  I know there is a case where we pass `NaT` instead of `iNaT`.  I think eventually we'll want to extend that method to handle this conceptually similar pattern which comes up a few times:

```
        if self.hasnans or other.hasnans:
            mask = (self._isnan) | (other._isnan)
            new_values[mask] = iNaT
```

Reply:
i would just change the default for now (or make it None and just fill with iNaT). i don't passing this directly (and IIRC it was never this way)

Reply:
can you change this

Reply:
done.